### PR TITLE
Fix/optimise dracut-install abs_rel

### PR DESCRIPTION
**Pull request by nabijaczleweli**
## Changes
Reduced allocations, fixes in patches 2 and 3

Split off #1796 

## Checklist
- [x] I have tested it locally
- [x] I have reviewed and updated any documentation if relevant
- [x] I am providing new code and test(s) for it

### CODE REVIEW

File: src/install/dracut-install.c [hunk 1]
```diff
@@ -162,39 +162,41 @@ static size_t dir_len(char const *file)
...
         /* dir_len() skips double /'s e.g. //lib64, so we can't skip just one
          * character - need to skip all leading /'s */
-        rl = strlen(target);
-        for (i = dirlen + 1; i < rl; ++i)
-                if (target_dir_p[i] != '/')
-                        break;
-        _asprintf(&realtarget, "%s/%s", realpath_p, &target_dir_p[i]);
+        for (i = dirlen + 1; target[i] == '/'; ++i)
+                ;
+        _asprintf(&realtarget, "%s/%s", buf, &target[i]);
```
Comment (marcosfrm) [inline src/install/dracut-install.c:195]: For consistency's sake `target + i`.

**Review by marcosfrm — COMMENTED** [review]
Nice cleanups. Commit messages need tweaking: full function name, component being modified.

**Comment by marcosfrm** [comment]
Forgot to say: mention in the commit message #1781 is closed by "feat: canonicalise parent of abs_rel() from, too".

**Comment by nabijaczleweli** [comment]
Applied all

**Review by marcosfrm — COMMENTED** [review]
Just minor stuff not code related: wrong bug number in the third commit (should be #1781); fourth commit message has a leftover "abs_rel()", which can be removed.

**Comment by nabijaczleweli** [comment]
applied

**Comment by marcosfrm** [comment]
LGTM

**Pull request merged**